src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/LambdaCompiler.Statements.cs

```diff
@@ -152,6 +152,28 @@ private void EmitLoopExpression(Expression expr)
         private void EmitSwitchExpression(Expression expr, CompilationFlags flags)
         {
             SwitchExpression node = (SwitchExpression)expr;
+            
+            if (node.Cases.Count == 0)
+            {
```

Comment:
You're pretty much hitting on the point @mikedn made in the opposite direction.

I added the reducibility later, with the hope that of consuming code that choked on case-less switches, maybe some of it attempts reducing, and as such doesn't actually get the expression that would cause it trouble.

I'd already changed up EmitSwitchExpression, and since EmitExpression doesn't attempt reducing on all nodes, it's still needed.

@mikedn  though points out that since reducing is now in there, changing stackspiller to remove its assumptions about what nodes can be reduced will mean this code becomes unreachable, and as such could be left unchanged.

And I think he's correct. I'm going to change this PR to match that approach.